tokio/src/net/tcp/socket.rs

```diff
@@ -482,6 +482,48 @@ impl TcpSocket {
         let mio = self.inner.listen(backlog)?;
         TcpListener::new(mio)
     }
+
+    /// Converts a TcpStream into a TcpSocket
+    /// This converts the TcpStream into
+    /// the platform specific raw_fd(unix) / raw_socket(windows)
+    /// and then converts them into the TcpSocket
+    /// as TcpSocket implements the FromRawFd(unix) and
+    /// FromRawSocket(windows) traits.
+
+    /// # Examples
+    /// ```
+    /// use tokio::net::TcpSocket;
+    ///
+    /// use std::net::TcpStream;
+    ///
+    /// #[tokio::main]
+    /// async fn main() -> io::Result<()> {
+    ///
+    ///     let stream = TcpStream::connect("127.0.0.1:8080")?;
+    ///
+    ///     let socket = TcpSocket::from_std_stream(stream)?;
+    /// # drop(socket);
+    ///
+    ///     Ok(())
+    /// }
+    /// ```
+    pub fn from_std_stream(std_stream: std::net::TcpStream) -> io::Result<TcpSocket> {
```

Comment:
The conversion always succeeds, so no need for a `Result`.
```suggestion
    pub fn from_std_stream(std_stream: std::net::TcpStream) -> TcpSocket {
```